core/ir/instr_api.h

```diff
@@ -722,6 +724,11 @@ DR_API
 int
 instr_get_opcode(instr_t *instr);
 
+DR_API
+/** Returns \p instr's set of categories (set of DR_INSTR_CATEGORY_ constants). */
```

Comment:
I only see categories being set on decode.  What if I modify a decoded instruction: I don't see the categories being invalidated.

Reply:
Added check that `instr_operands_valid(instr))` is true. redecode if operands are invalid